src/amp-story-embed.js

```diff
@@ -123,34 +194,145 @@ export class AmpStoryEmbed {
   }
 
   /**
-   * @return {!Promise}
    * @public
    */
   layoutCallback() {
-    // TODO: Layout all child iframes.
-    return this.layoutIframe_(0);
+    if (this.isLaidOut_) {
+      return;
+    }
+
+    const iframes = toArray(this.rootEl_.querySelectorAll('iframe'));
+    // TODO(Enriqe): Layout all child iframes.
+    this.layoutIframe_(this.stories_[0], iframes[0]);
+
+    this.isLaidOut_ = true;
   }
 
   /**
-   * @param {number} index
-   * @return {!Promise}
+   * @param {!Element} story
+   * @param {!Element} iframe
    * @private
    */
-  layoutIframe_(index) {
-    const story = this.stories_[index];
-    this.iframeEl_.setAttribute('src', story.href);
-    return Promise.resolve();
+  layoutIframe_(story, iframe) {
+    const {href} = this.getEncodedUrl_(story.href);
+
+    iframe.setAttribute('src', href);
+  }
+
+  /**
+   * Gets encoded url for viewer usage.
+   * @param {string} href
+   * @return {!Location}
+   * @private
+   */
+  getEncodedUrl_(href) {
```

Comment:
Please update the name of this method or the returned type, `URL` and `Location` are different objects